### PR TITLE
Complete Use of DataFactory with termFromId

### LINKED ISSUE #222: Missing a Use of Custom DataFactory in N3 Store
The `termFromId` function from the `DataFactory` is called with the resolved DataFactory implemention of the N3 Store in all cases except the one in `_uniqueEntities`. This makes it so that the terms in the store could be a mix of N3 DataFactory terms and the custom DataFactory terms.

This is a simple fix. I'll work on a PR for this.

### PR DESCRIPTION
**Pull request by dhurlburtusa**
Ensure DataFactory, potentially a custom implementation, is used throughout the store.

Fixes #222.

### CODE REVIEW

**Comment by dhurlburtusa** [comment]
I am not sure what the issue is with the failed test. Please let me know if there is something I need to do on my end.

Note: All unit tests in all environments pass.
Note: All specs in all environments pass except one.

Seems like there is a missing build artifact that causes the failure.

**Comment by RubenVerborgh** [comment]
@dhurlburtusa Thanks!

@rubensworks Seems like a failure of rdf-test-suite on Node 15?

File: src/N3Store.js [hunk 1]
```diff
@@ -198,11 +198,11 @@ export default class N3Store {
   // ### `_uniqueEntities` returns a function that accepts an entity ID
   // and passes the corresponding entity to callback if it hasn't occurred before.
   _uniqueEntities(callback) {
-    var uniqueIds = Object.create(null), entities = this._entities;
+    var uniqueIds = Object.create(null), entities = this._entities, factory = this._factory;
```
Comment (RubenVerborgh) [inline src/N3Store.js:201]: I maybe wonder if we want to convert the below to an arrow function instead? Then we don'tt need to store all the variables.

**Comment by rubensworks** [comment]
@RubenVerborgh I assume it's a Node 15 bug (given that it's just out).
It's failing here: https://github.com/rdfjs/N3.js/blob/master/spec/parser.js#L1
Which should be correct.

**Comment by RubenVerborgh** [comment]
I suspect 15 has a different module system; so maybe not a bug.

File: src/N3Store.js (hunk 1 above)
Comment (dhurlburtusa) [reply src/N3Store.js:201]: I'd be happy to make another PR and change all the nested functions to arrow functions and remove all the `var self = this;` uses and other unnecessary closure creations.

Here is list of some of the areas that would be updated.

- N3Lexer.js in tokenize
- N3Parser.js in parse
- N3Store in _uniqueEntities, import, and remove
- N3StreamParser constructor and import
- N3StreamWriter constructor and import

There may be others too.

Comment (RubenVerborgh) [reply src/N3Store.js:201]: @dhurlburtusa Thanks, that would be nice indeed! Could you change this single one already for this PR? Then I will merge.

Comment (dhurlburtusa) [reply src/N3Store.js]: Updated.

**Comment by RubenVerborgh** [comment]
Rebased to circumvent the Node 15 issue (#229).

**Pull request merged**